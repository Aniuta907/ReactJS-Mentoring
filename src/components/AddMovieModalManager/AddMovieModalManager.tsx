import React, { useState } from "react";

import { ModalWindow } from "../ModalWindow";
import { AddMovieModalContent } from "../AddMovieModalContent";
import "./AddMovieModalManager.scss";

export const AddMovieModalManager: React.FC = () => {
  const [IsModalVisible, setModalVisibility] = useState(false);

  const openModal = () => {
    setModalVisibility(true);
  };

  const closeModal = () => {
    setModalVisibility(false);
  };

  return (
    <>
      <button className="add-movie-button" onClick={openModal}>
        + add movie
      </button>
      {IsModalVisible && (
        <ModalWindow
          IsModalVisible={IsModalVisible}
          modalTitle="ADD MOVIE"
          leftButton="RESET"
          rightButton="SUBMIT"
          closeModal={closeModal}
        >
          <AddMovieModalContent />
        </ModalWindow>
      )}
    </>
  );
};
