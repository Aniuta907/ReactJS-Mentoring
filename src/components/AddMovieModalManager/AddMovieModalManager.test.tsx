import * as React from "react";
import { mount, shallow, ShallowWrapper } from 'enzyme';

import { AddMovieModalManager } from './AddMovieModalManager';

describe('AddMovieModalManager', () => {
    it('should render correctly', () => {      
        const wrapper: ShallowWrapper = shallow(<AddMovieModalManager />);
        expect(wrapper).toMatchSnapshot();
    });

    test('AddMovieModalManager sets state properly', () => {
        const setModalVisibility = jest.fn();
        const wrapper = mount(<AddMovieModalManager />);
        const handleClick = jest.spyOn(React, "useState");
        // Type 'any[]' is not assignable to type '[unknown, Dispatch<unknown>]'. 
        // Target requires 2 element(s) but source may have fewer
        handleClick.mockImplementation(IsModalVisible => [IsModalVisible, setModalVisibility]);

        expect(setModalVisibility).toBeFalsy();
        wrapper.find(".add-movie-button").simulate("click");
        expect(setModalVisibility).toBeTruthy();
      })
});
