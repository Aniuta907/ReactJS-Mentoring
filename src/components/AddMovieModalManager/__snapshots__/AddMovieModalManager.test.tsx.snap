// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovieModalManager should render correctly 1`] = `
<Fragment>
  <button
    className="add-movie-button"
    onClick={[Function]}
  >
    + add movie
  </button>
</Fragment>
`;
